lib/tapioca/dsl/compilers/controller_actions.rb

```diff
@@ -0,0 +1,75 @@
+# typed: strict
+# frozen_string_literal: true
+
+begin
+  require "action_controller"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Dsl
+    module Compilers
+      # `Tapioca::Dsl::Compilers::ControllerActions` generates RBI files for classes that include
+      # [`ActionController::Base`](https://api.rubyonrails.org/classes/ActionController/Base.html).
+      #
+      # For example, with the following class that includes `ActionController::Base`:
+      #
+      # ~~~rb
+      # class PostsController < ApplicationController
+      #  def index
+      #   # ...
+      #  end
+      #
+      #  def create
+      #    # ...
+      #  end
+      # end
+      # ~~~
+      #
+      # this compiler will produce the RBI file `posts_controller.rbi` with the following content:
+      #
+      # ~~~rbi
+      # # posts_controller.rbi
+      # # typed: strong
+      # class PostsController
+      #   sig { void }
+      #   def index; end
+      #
+      #   sig { void }
+      #   def create; end
+      # end
+      # ~~~
+      class ControllerActions < Compiler
+        extend T::Sig
+
+        ConstantType = type_member { { fixed: T.class_of(::ActionController::Base) } }
+
+        sig { override.void }
+        def decorate
+          public_methods = constant.instance_methods(false)
```

Comment:
This returns `protected` methods too and [I don't think we want that](https://guides.rubyonrails.org/action_controller_overview.html#:~:text=Only%20public%20methods%20are%20callable%20as%20actions.%20It%20is%20a%20best%20practice%20to%20lower%20the%20visibility%20of%20methods%20(with%20private%20or%20protected)%20which%20are%20not%20intended%20to%20be%20actions%2C%20like%20auxiliary%20methods%20or%20filters.).

Reply:
Can we also add a test with `protected` methods?